### PR TITLE
DG - Write Design and Implementation of File Management

### PR DESCRIPTION
**Pull request by alanlowzies**
This PR contains my (very long) write up of the design and implementation of the storage component of WerkIt!

## Note
For the section on 'Implementation - Storage Format for Each Resource File', the table formatting is very weird because of the pipe symbol. However, on GitHub Pages it looks fine.

**Raw Code:**
image

**GitHub:**
image

**GitHub Pages;**
image

Other changes include:
- DG - Added target user profile and value proposition
- PPP - Added this PR's write up to my DG contribution list
- Add static buttons around the DG and UG to improve navigability.

View the PR in GitHub Pages [here](https://alanlowzies.github.io/tp/DeveloperGuide.html)

### CODE REVIEW

**Comment by alanlowzies** [comment]
The 'buttons' should not be placed directly below tables as they clip each other.

**Review by emilysim00 — APPROVED** [review]
Love the idea of adding the button for better navigation around the document. Thanks !

File: docs/uml/classDiagrams/StorageComponent.puml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    + printExerciseFileCreatedMessage() : void
+}
+
+class WerkIt
+
+package "Command Component" <<rectangle>> {
+    class WorkoutCommand
+    class PlanCommand
+    class ScheduleCommand
+}
```
Comment (Musfirahe0556596) [inline docs/uml/classDiagrams/StorageComponent.puml:25]: Changed naming of Command --> Logic component x)

**Review by Musfirahe0556596 — APPROVED** [review]
Just a minor naming change

File: docs/uml/classDiagrams/StorageComponent.puml (hunk 1 above)
Comment (alanlowzies) [reply docs/uml/classDiagrams/StorageComponent.puml]: Informed via Telegram. Resolved in Commit 4e71996.

**Pull request merged**